### ISSUE TITLE
Error in anonymous login and its fix.

### ISSUE THREAD

**Issue by liangdefeng** [esl/MongooseIM#3703]
MongooseIM version: 5.1.0
Installed from: source or pkg
Erlang/OTP version: 23.3.4.15

Describe the issue.

 - Error occurs when anonymous is configured in auth.

mongooseim.toml 

```
...
[auth]
  methods = ["rdbms","anonymous"]
  sasl_mechanisms = ["scram_sha1","anonymous"]
  password.hash = ["sha"]
  password.scram_iterations = 5_000

  [auth.rdbms]
    users_number_estimate = true

  [auth.anonymous]
...

```

When use Spark to do anonymous login, the error appears in the server's log.

```

level=error what=hook_failed reason="{function_clause,[{ejabberd_auth,'-does_stored_user_exist/2-fun-0-',[ejabberd_auth_anonymous],[{file,\"/root/MongooseIM/src/auth/ejabberd_auth.erl\"},{line,326}]},{ejabberd_auth,fold_auth_modules,3,[{file,\"/root/MongooseIM/src/auth/ejabberd_auth.erl\"},{line,552}]},{timer,tc,2,[{file,\"timer.erl\"},{line,181}]},{ejabberd_auth,call_auth_modules_for_host_type,3,[{file,\"/root/MongooseIM/src/auth/ejabberd_auth.erl\"},{line,513}]},{ejabberd_auth,does_user_exist,4,[{file,\"/root/MongooseIM/src/auth/ejabberd_auth.erl\"},{line,314}]},{ejabberd_hooks,gen_hook_fn_wrapper,3,[{file,\"/root/MongooseIM/src/ejabberd_hooks.erl\"},{line,109}]},{safely,apply,2,[{file, MongooseIM\"/root/MongooseIM/src/safely.erl\"},{line,42}]},{gen_hook,run_hook,4,[{file,\"/root/MongooseIM/src/gen_hook.erl\"},{line,203}]}]}" pid=<0.1621.0> at=gen_hook:log_error_running_hook/5:227 text="Error running hook" params_args="[<<\”example.com\”>>,{jid,<<\"f9c0382ce23123c04\">>,<<\”example.com\”>>,<<\"8ab284e384adf8a61657-350403-175477\">>},stored]" key="{does_user_exist,<<\”example.com\”>>}" handler="{hook_handler,50,fun ejabberd_hooks:gen_hook_fn_wrapper/3,#{function => does_user_exist,hook_name => does_user_exist,hook_tag => <<\"example.com\">>,host_type => <<\"example.com\">>,module => ejabberd_auth}}" 

```

The following change can fix the issue.

File to change: ejabberd_auth.erl

Change Before
```
...
does_stored_user_exist(HostType, #jid{luser = LUser, lserver = LServer}) ->
    F = fun(Mod) when Mod =/= ejabberd_auth_anonymous ->
                does_user_exist_in_module(HostType, LUser, LServer, Mod)
       end,
...

```

Change After

```
...
does_stored_user_exist(HostType, #jid{luser = LUser, lserver = LServer}) ->
    F = fun(Mod) ->
          case Mod of
            ejabberd_auth_anonymous ->
              {stop, false};
            _ ->
              does_user_exist_in_module(HostType, LUser, LServer, Mod)
          end
        end,
    call_auth_modules_for_host_type(HostType, F, #{default => false, metric => does_user_exist});
...

```

**Comment by chrzaszcz** [comment]
Thanks for the report. The bug is fixed by #3706.

**Comment by liangdefeng** [comment]
Thanks for the fix.
